Homework 3

Pashacabu opened the pull request:
1.MovieDetails.xml moved to MovieDetails.kt fragment as a layout to inflate
2.Created activity_main.xml with FrameLayout as a container for fragments
3.Created MoviesList.kt fragment with its own layout as show in Figma
4.Attached MoviesList to a fragment container as default
5.Added ClickListener to open MovieDetails fragment over MoviesList with adding MovieDetails fragment to backstack
6.Added ClickListener to mark movie from MoviesList as "liked"
7.Added ClickListener to go back in stack (to the MoviesList) when back arrow is pressed in MoviesDetails

Screen rotation is working correctly, instance state is saving. The only thing is missing is saving "Liked" button state when app is reopened but I'm not sure it's needed at the moment

![Без имени-1](https://user-images.githubusercontent.com/70441207/100475417-1bb60800-30f4-11eb-9d68-31029d208048.jpg)

Pashacabu committed:
Correction

Pashacabu committed:
Saving state for Liked button fixed

PaulXIII commented on app/src/main/java/com/pashcabu/hw2/MainActivity.kt:
```diff
@@ -4,20 +4,37 @@ import androidx.appcompat.app.AppCompatActivity
+                    .add(R.id.fragment_container, fragmentMoviesList, "MoviesList")
+                    .commit()
+        } else {
+            fragmentMoviesList=supportFragmentManager.findFragmentByTag("MoviesList") as MoviesList
```
Строка "MoviesList" используется в нескольких местах. Лучше сделать это значения константой

PaulXIII commented on app/src/main/java/com/pashcabu/hw2/MainActivity.kt:
```diff
@@ -4,20 +4,37 @@ import androidx.appcompat.app.AppCompatActivity
-        val btnBack : ImageView = findViewById(R.id.backArrow)
-        btnBack.setOnClickListener(this)
+        setContentView(R.layout.activity_main)
+        fragmentContainer=findViewById(R.id.fragment_container)
```
`FrameLayout` в этом примере испльзуется как контейнер для `Fragment`.
Только для этой цели нет нужды использвать `findViewById`

PaulXIII commented on app/src/main/java/com/pashcabu/hw2/MainActivity.kt:
```diff
@@ -4,20 +4,37 @@ import androidx.appcompat.app.AppCompatActivity
+    override fun openMovieDetails() {
+        supportFragmentManager.beginTransaction()
+                .add(R.id.fragment_container, fragmentMovieDetails, "MovieDetails")
+                .addToBackStack("Movie Details")
```
Аналогичная ситуация с константой

PaulXIII commented on app/src/main/java/com/pashcabu/hw2/MoviesList.kt:
```diff
@@ -0,0 +1,80 @@
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putBoolean("isLiked", isLiked)
```
Значение "isLiked" используется в нескольких местах. Также лучше сделать константу

PaulXIII commented on app/src/main/res/drawable/movie_card_background.xml:
```diff
@@ -0,0 +1,17 @@
+    <corners android:radius="8dp"/>
+    <stroke
+        android:width="1dp"
+        android:color="#80636375"/>
```
Значения цветов лучше перенести в `colors.xml` и вызывать по значению

PaulXIII commented on app/src/main/res/layout/movie_details_fragment.xml:
```diff
@@ -0,0 +1,349 @@
+            android:layout_marginTop="8dp"
+            android:src="@drawable/star"
+            app:layout_constraintLeft_toRightOf="@id/star_4"
+            app:layout_constraintTop_toBottomOf="@id/tag_line" />
```
Отображение рейтинга можно сделать с использованием компонента `RatingBar`.
Пример как использовать https://www.tutorialspoint.com/how-to-use-ratingbar-in-android

Pashacabu committed:
Deleted unnecessary findViewById call
Moved hardcoded strings to constants
Corrected layout to inflate for MovieDetails
Moved hardcoded colors to colors.xml
Used RatingBar for movie rating
